kbase-extension/static/kbase/js/common/spec.js

```diff
@@ -59,14 +62,38 @@ define(['bluebird', 'common/lang', 'common/sdk', 'widgets/appWidgets2/validators
             return model;
         }
 
+        /**
+         * Validates the model (object that maps from parameter id to current value) against
+         * this spec. This returns a Promise that resolves into a map from parameter ids to
+         * validations.
+         * @param {object} model the object containing the data model to validate
+         * should have key-value-pairs for each parameter id.
+         * @returns Promise that resolves into a mapping from parameter id -> validation
+         * structure
+         */
         function validateModel(model) {
-            // TODO: spec at the top level should be a struct...
-            // return;
+            return validateParams(spec.parameters.layout, model);
+        }
+
+        /**
+         * A trimmed version of validateModel that's specific for a few params.
+         * Given an array of parameter ids and an object with key-value-pairs from
+         * paramId -> value, validate the set. Only the given parameter ids are validated.
+         * Any others are ignored.
+         *
+         * This returns a Promise that resolves into the a key-value-pair of parameter id ->
```

Comment:
`the key-value-pair` or `a key-value-pair`?

Punctuationally speaking it should probably be `key-value pair`

Reply:
Yes. Fixed punctuation.